Contains fix for bug #9

kspviswa opened the pull request:
# Contains fix for bug #9 .

## Explanation

Email-ID validation can be done at client itself. In-order to support it, I had extended `validate_fields()` in `forms.js` file. The given email-id will be compared with valid format and if found correct, the form will be submitted. Else a error string will be displayed. In order to support the error string, `errors.html.template` has been modified and new `<div>` element has been introduced. Basic unit testing has been completed. 

### Sample test result
![image](https://cloud.githubusercontent.com/assets/7476271/7419696/79cdc2f4-ef94-11e4-85b5-8f28d9f41144.png)

kspviswa commented:
This is my first pull request. Kindly guide me, if anything is wrong in what I'm doing

skywhale commented on app/resources/errors.html.template:
```diff
@@ -8,6 +8,10 @@
   </div>
+  <div id="email_id_improper_format"
+      class="mandatory" style="display: none">
+    {% trans "The Email-ID you entered doesn't look like valid. Please correct it and try again" %}
```
s/Email-ID/Email address/
plz also add a period at the end of the sentence.

skywhale commented on app/resources/forms.js:
```diff
@@ -284,6 +284,22 @@ function validate_fields() {
+  var email_field = $('author_email');
+  var email_id = $('author_email').value;
+  if (email_id !== "")
+  {
```
please follow the formatting convention in the file.

e.g.

if (email_id !== '') {
  ...
}

skywhale commented on app/resources/forms.js:
```diff
@@ -284,6 +284,22 @@ function validate_fields() {
+  var email_id = $('author_email').value;
+  if (email_id !== "")
+  {
+     var emailfilter=/^\w+[\+\.\w-]*@([\w-]+\.)*\w+[\w-]*\.([a-z]{2,4}|\d+)$/i;
```
Let's use the same regex as the server-side:
https://github.com/google/personfinder/blob/master/app/subscribe.py#L26

kspviswa commented:
@skywhale Thanks for your valuable suggestions. I had recommitted the changes addressing your comments. 

*note* : Do I have to create another pull request for this review. I'm new to github branching & reviewing. Please guide me.

skywhale commented on app/resources/forms.js:
```diff
@@ -284,6 +284,20 @@ function validate_fields() {
+  var email_field = $('author_email');
+  var email_id = $('author_email').value;
+  if (email_id !== ""){
+     var emailfilter=/(?:^|\s)[-a-z0-9_.%$+]+@(?:[-a-z0-9]+\.)[a-z]{2,6}(?:\s|$)/i;
```
missing "+" after "(?:[-a-z0-9]+\.)"

skywhale commented on app/resources/errors.html.template:
```diff
@@ -8,6 +8,13 @@
+  <div id="email_id_improper_format"
+      class="mandatory" style="display: none">
+  {% blocktrans %}
+  The Email address you entered doesn't look like valid address.
```
To shorten it and make localization easier.

"The Email address you entered is not valid."

skywhale commented:
Thank you!

Please correct the regex and the message, and it's good to go. We will merge the pull request and deploy it with couple other changesets.

kspviswa commented:
@skywhale  pushed the changes as suggested.
